File: src/utils/dateUtils.js
export const formatDisplayDate = (value) => {
  const date = new Date(value);
  return new Intl.DateTimeFormat('en-US', {
    weekday: 'long',
    month: 'long',
    day: 'numeric',
  }).format(date);
};

export const formatTime = (date) =>
  new Intl.DateTimeFormat('en-US', {
    hour: 'numeric',
    minute: '2-digit',
  }).format(date);

export const dateKeyFromDate = (date) => new Date(date).toISOString().slice(0, 10);

export const toDateFromTask = (dateString, timeString) => {
  const [hours, minutes] = timeString.split(':').map(Number);
  const next = new Date(dateString);
  next.setHours(hours, minutes, 0, 0);
  return next;
};

File: lib/utils/date_utils.dart
import 'package:intl/intl.dart';

class AppDateUtils {
  static String formatDateKey(DateTime date) => DateFormat('yyyy-MM-dd').format(date);

  static String formatTime12h(String time24) {
    final parts = time24.split(':');
    final hour = int.parse(parts[0]);
    final minute = int.parse(parts[1]);
    final period = hour >= 12 ? 'PM' : 'AM';
    final hour12 = hour % 12 == 0 ? 12 : hour % 12;
    return '$hour12:${minute.toString().padLeft(2, '0')} $period';
  }
}